src/DblParser/Desugar.ml

```diff
@@ -816,4 +837,45 @@ and make_handle ~pos lbl_opt eff_opt pat body hcs =
       fin_clauses = fcs
     })
 
-let tr_program (p : Raw.program) = { p with data = tr_defs p.data }
+and generate_accessor_method_pattern named_type_args type_name =
+  let make data = { data; pos=Position.nowhere } in
+  let create_mapping i arg =
+    let old_name, new_name =
+      match (snd arg.data).data with
+      | TA_Var(name, _) -> name, name ^ "#TA_Var#" ^ string_of_int i
+      | TA_Effect -> "", "#TA_Effect#" ^ string_of_int i
+      | TA_Wildcard -> "", "#TA_Wildcard#" ^ string_of_int i
+    in
+    make (TNVar old_name, make (TA_Var(new_name, make (KWildcard)))), new_name
+  in
+  let new_named_type_args, new_names = List.split (List.mapi create_mapping named_type_args) in
+  let type_annot : scheme_expr =
+    { sch_pos = Position.nowhere
+    ; sch_targs = []
+    ; sch_named = []
+    ; sch_body = List.fold_left
+      (fun inner name -> make (TApp(inner, make (TVar(NPName name)))))
+      (make (TVar(NPName type_name)))
+      new_names
+    } in
+  let pattern_gen field =
+    ArgAnnot(make (PCtor(
+      make (NPName type_name),
+      [],
+      [make (NVar field, make (PId(IdVar(false, field))))],
+      [])), type_annot) in
+  new_named_type_args, pattern_gen
+
+and create_accessor_method ~public named_type_args pattern_gen scheme =
```

Comment:
This and the previous function generate some piece of code, which is hard to read directly in the abstract syntax. Maybe a short documentation comment with a simple example using concrete syntax would help.